marge/job.py

```diff
@@ -182,14 +188,30 @@ def wait_for_ci_to_pass(self, merge_request, commit_sha=None):
             if ci_status == 'canceled':
                 raise CannotMerge('Someone canceled the CI.')
 
-            if ci_status not in ('pending', 'running'):
+            if ci_status == 'manual':
+                if play_manual_jobs:
+                    self.play_pending_manual_jobs(merge_request, pipeline.id)
+                else:
+                    raise CannotMerge('Manual Step encountered with play_manual_jobs set to False')
+            if ci_status not in ('running', 'pending', 'manual'):
                 log.warning('Suspicious CI status: %r', ci_status)
 
             log.debug('Waiting for %s secs before polling CI status again', waiting_time_in_secs)
             time.sleep(waiting_time_in_secs)
 
         raise CannotMerge('CI is taking too long.')
 
+    def play_pending_manual_jobs(self, merge_request, pipeline_id):
+        manual_job_list = Pipeline.manual_jobs_by_pipeline(
+            merge_request.target_project_id,
+            pipeline_id,
+            self._api
+        )
```

Comment:
Gitlab CI supports manual actions that don't need to be run for the pipeline to succeed, and ones that must run and succeed, via the [allow_failure](https://docs.gitlab.com/ee/ci/yaml/#allow_failure) option. Any chance we could have an option to run only the manual jobs which are `allow_failure=false` ?